aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/SynchronousNonBlockingStepTest.java

```diff
@@ -0,0 +1,250 @@
+            "echo 'First message'\n" +
+            "syncnonblocking 'wait'\n" +
+            "echo 'Second message'\n" +
+        "}"));
+        WorkflowRun b = p.scheduleBuild2(0).getStartCondition().get();
+
+        // Wait for syncnonblocking to be started
+        System.out.println("Waiting to syncnonblocking to start...");
+        SynchronousNonBlockingStep.waitForStart("wait", b);
+
+        // At this point the execution is paused inside the synchronous non-blocking step
+        // Check for FlowNode created
+        FlowGraphWalker walker = new FlowGraphWalker(b.getExecution());
+        boolean found = false;
+        // TODO: use iterator when https://github.com/jenkinsci/workflow-plugin/pull/178 merged
+        for (FlowNode n = walker.next(); n != null; n = walker.next()) {
+            if (n instanceof StepNode && ((StepNode) n).getDescriptor() instanceof SynchronousNonBlockingStep.DescriptorImpl) {
+                found = true;
+                break;
+            }
+        }
+
+        System.out.println("Checking flow node added...");
+        assertTrue("FlowNode has to be added just when the step starts running", found);
+
+        // Check for message the test message sent to context listener
+        System.out.println("Checking build log message present...");
+        j.waitForMessage("Test Sync Message", b);
+        // The last step did not run yet
+        j.assertLogContains("First message", b);
+        j.assertLogNotContains("Second message", b);
+
+        // Let syncnonblocking to continue
+        SynchronousNonBlockingStep.notify("wait");
+
+        System.out.println("Waiting until syncnonblocking (and the full flow) finishes");
+        j.waitForCompletion(b);
+        System.out.println("Build finished. Continue.");
+        // Check for the last message
+        j.assertLogContains("Second message", b);
+        j.assertBuildStatusSuccess(b);
+    }
+
+    @Test
+    public void interruptedTest() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("node {\n" +
+            "echo 'First message'\n" +
+            "try { syncnonblocking 'wait' } catch(InterruptedException e) { echo 'Interrupted!' }\n" +
+            "echo 'Second message'\n" +
+        "}"));
+        WorkflowRun b = p.scheduleBuild2(0).getStartCondition().get();
+
+        // Wait for syncnonblocking to be started
+        System.out.println("Waiting to syncnonblocking to start...");
+        SynchronousNonBlockingStep.waitForStart("wait", b);
+
+        // At this point syncnonblocking is waiting for an interruption
+
+        FlowExecution e = b.getExecutionPromise().get();
+        // Let's force a call to stop. This will try to send an interruption to the run Thread
+        e.interrupt(Result.ABORTED);
+        System.out.println("Waiting for interruption");
+        j.waitForCompletion(b);
```

Comment:
:ant: you used to be waiting for a message here, but deleted that for some reason. As written you are not really testing much, only that the build ends for some reason. Since there is a `catch` block, I think the expectation is that you want the build to succeed, and to print the `Interrupted!` message.